### PR TITLE
[SPARK-25249][CORE][TEST]add a unit test for OpenHashMap

### PR DESCRIPTION
**Pull request by 10110346**
## What changes were proposed in this pull request?

This PR adds a unit test for OpenHashMap , this can help developers  to distinguish between the 0/0.0/0L and null

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/util/collection/OpenHashMapSuite.scala [hunk 1]
```diff
@@ -194,4 +194,42 @@ class OpenHashMapSuite extends SparkFunSuite with Matchers {
...
+
+  test("distinguish between the 0/0.0/0L and null") {
+    val specializedMap1 = new OpenHashMap[String, Long]
+    specializedMap1("a") = null.asInstanceOf[Long]
+    specializedMap1("b") = 0L
+    assert(specializedMap1.contains("a"))
+    assert(!specializedMap1.contains("c"))
+    assert(Some(specializedMap1("a")).contains(0L))
+    assert(Some(specializedMap1("b")).contains(0L))
+    assert(Some(specializedMap1("c")).contains(0L))
```
Comment (maropu) [inline core/src/test/scala/org/apache/spark/util/collection/OpenHashMapSuite.scala]: What do these tests mean? Tests for different type comparisons? e.g.,
```
    assert(!Some(specializedMap1("a")).contains(0.0))
    assert(!Some(specializedMap1("b")).contains(0.toShort))
```

**Comment by maropu** [comment]
plz add `[CORE]` in the title?

File: core/src/test/scala/org/apache/spark/util/collection/OpenHashMapSuite.scala (hunk 1 above)
Comment (10110346) [reply core/src/test/scala/org/apache/spark/util/collection/OpenHashMapSuite.scala]: if the data type is in `@specialized annotation`  , and the `key` not be contained ,the `map(key)` will  return 0
if the data type is not  in `@specialized annotation`  , and the `key` not be contained ,the `map(key)` will  return null

**Comment by 10110346** [comment]
thanks @maropu

File: core/src/test/scala/org/apache/spark/util/collection/OpenHashMapSuite.scala (hunk 1 above)
Comment (10110346) [reply core/src/test/scala/org/apache/spark/util/collection/OpenHashMapSuite.scala]: ok , i will add it

**Review by srowen — APPROVED** [review]

**Comment by kiszk** [comment]
Adding tests looks good.
Probably beyond the purpose of this PR, is the following difference intentionally designed?

```
    // If the data type is in @specialized annotation, and
    // the `key` is not be contained, the `map(key)` will return 0
```
```
    // If the data type is not in @specialized annotation, and
    // the `key` is not be contained, the `map(key)` will return null
```

**Review by HyukjinKwon — APPROVED** [review]

**Comment by srowen** [comment]
@kiszk I guess it's because in this case the underlying value type is a primitive like int or long, so null can't be returned?

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**